tests/testthat/test-recover-weaker.R

```diff
@@ -0,0 +1,174 @@
+                             rt_crosstab = as_feature_crosstab(feature_names, sample_names, rt_crosstab),
+                             int_crosstab = as_feature_crosstab(feature_names, sample_names, int_crosstab)
+    )
+    
+    aligned_feature_sample_table_actual <- create_feature_sample_table(aligned)
+    recovered_feature_sample_table_actual <- create_feature_sample_table(recovered_actual)
+    
+    aligned_feature_sample_table_expected <- arrow::read_parquet(file.path(testdata, "recovered", 'aligned_feature_sample_table.parquet'))
+    recovered_feature_sample_table_expected <- arrow::read_parquet(file.path(testdata, "recovered", 'recovered_feature_sample_table.parquet'))
+    
+    expect_equal(aligned_feature_sample_table_actual, aligned_feature_sample_table_expected)
+    expect_equal(recovered_feature_sample_table_actual, recovered_feature_sample_table_expected)
+    
+    # create and load final files
+    
+    extracted_recovered_actual <- recovered_actual$extracted_features
+    corrected_recovered_actual <- recovered_actual$corrected_features
+    
+    filenames <- lapply(files, function(x) {
+      file.path(testdata, "recovered", "recovered-extracted", paste0(x, ".parquet"))
+    })
+    
+    extracted_recovered_expected <- lapply(filenames, arrow::read_parquet)
+
+    filenames <- lapply(files, function(x) {
+      file.path(testdata, "recovered", "recovered-corrected", paste0(x, ".parquet"))
+    })
+    
+    corrected_recovered_expected  <- lapply(filenames, arrow::read_parquet)
+    
+    # preprocess dataframes
+    keys <- c("mz", "pos", "sd1", "sd2")
+    
+    extracted_recovered_actual <- lapply(extracted_recovered_actual, function(x) {
+      as.data.frame(x) |> dplyr::arrange_at(keys)
+    })
+    corrected_recovered_actual <- lapply(corrected_recovered_actual, function(x) {
+      as.data.frame(x) |> dplyr::arrange_at(keys)
+    })
+    
+    extracted_recovered_expected <- lapply(extracted_recovered_expected, function(x) {
+      as.data.frame(x) |> dplyr::arrange_at(keys)
+    })
+    corrected_recovered_expected <- lapply(corrected_recovered_expected, function(x) {
+      as.data.frame(x) |> dplyr::arrange_at(keys)
+    })
+    
+    # compare files
+    for (i in seq_along(files)) {
+      # extracted recovered
+      actual_extracted_i <- extracted_recovered_actual[[i]]
+      expected_extracted_i <- extracted_recovered_expected[[i]]
+      
+      report <- dataCompareR::rCompare(actual_extracted_i, expected_extracted_i, keys = keys, roundDigits = 4, mismatches = 100000)
+      dataCompareR::saveReport(report, reportName = files[[i]], showInViewer = FALSE, HTMLReport = FALSE, mismatchCount = 10000)
+      
+      expect_true(nrow(report$rowMatching$inboth) >= 0.9 * nrow(expected_extracted_i))
+      
+      incommon <- as.numeric(rownames(report$rowMatching$inboth))
+      
+      subset_actual <- actual_extracted_i %>% dplyr::slice(incommon)
```

Comment:
Wouldn't it make more sense to use the same pipe throughout the whole script? Or `|>` won't work here?